### ISSUE TITLE
Organsystem additions

### ISSUE THREAD

**Issue by lkotipal** [octareenroon91/-tg-station#14]
Moving this issue. Also, post any glitches and problems regarding dismemberment here
- [ ] More internal organs, like liver, kidneys and stomach. Some internal organs have diseases linked to them so medbay might want to keep an organ bank.
  - [x] Livers and cirrhosis caused by excessive alcohol use (20% chance on alcohol poisoning)
  - [x] Kidneys and renal failure caused by large amounts of toxins (>=50 toxloss)
  - [x] Lungs and emphysema caused by nicotine (5% chance on stage 1 addiction)
  - [ ] Arrhythmia
  - [ ] Stomach: can't keep food down if missing
- [ ] Internal organs actually doing something when not fucked: toxin filtering etc.
  - [ ] Upgradable internal organs for SPESS MUHRENS
- [ ] Ligger tail removal and limb regeneration, the latter controlled by an internal organ secreting a chemical
- [ ] Broken limbs and ways to treat them such as crutches, splints and wheelchairs
- [ ] Draw 'X' on unusable inventory slots
- [x] Chainsaw arms, peg legs and other silliness
- [ ] Move sight procs to eyes and fix alium darksight eyes on humans
- [ ] Tongue or vocal chords as an organ. Their DNA would determine your voice, and they could be manipulated with a scalpel to produce a different one.
- [ ] Mutations controlled by organs
- [ ] Actual limbs with locational damage for all carbons (except slimes. Why are they carbons again?)
  - [ ] Porting over DNA related fixes from upstream so monkeys have DNA (https://github.com/tgstation/-tg-station/pull/11883/ and https://github.com/tgstation/-tg-station/pull/11963)

**Comment by octareenroon91** [comment]
> give lizard spec_life a chance to re-generate missing limbs
> you can grow limbs back, but only if you're a lizard
> Dr_Connors.png

**Comment by Koriath** [comment]
Was gonna post a diferent issue, but I think it fits here:
I found out why monkeys don't yield organs.
When you turn a monkey into a human to harvest the limbs for instance, the organsystem doesn't assign an owner to the limbs/organs:
http://puu.sh/ngk9m/8997f315c7.png
left is a spawned monkey turned human, right is a regular spawned human.
Might wanna look into this.

**Comment by lkotipal** [comment]
Good find, looks like transform procs don't set ownership properly. I'll push an attempted fix right away.

**Comment by Koriath** [comment]
> Internal organs actually doing something when not fucked: toxin filtering etc
> I got a few simple sugestions:
> Kydneys: remove Tox damage overtime (slowly tough)
> Lungs: Remove O2 damage overtime (instead of flat amount, it's dependant on lung type/damage)
> Heart: Stamina recovery. Some advanced hearts could add hyperzine to the holder. Or cofee.
> Liver: Blood regeneration rate, since it's the liver that produces most of it.

These are simple to implement I think. Most of these values are static right now, we could simply read them off from the organs instead of using a DEFINE for them.

**Comment by Koriath** [comment]
ASD, there's some investigation I've been doing on the "invisible races" for which I sent the sprites in.
Just out of curiosity: the one's I sent didn't have diferent chests for male and female. Could that be the cause? I haven't looked too much into the code, just looking at varediting humans.

The "renderkey" var looks like one giant string that's constructed with code. What would happen if it was borked, or didn't match existing sprite names? Would the whole thing break and refuse to draw the body at all?

I'm gonna attempt a small fix on my computer first, but I'd like your opinion.
There's also a weird bug on lizards. They aparently draw the body, but they don't respect the "green color" thing. Except for the tail, which is in fact green, but only when the sprite is facing north.
Any idea on why that happened?

**Comment by lkotipal** [comment]
IIRC renderkey doesn't actually draw the sprite, it's just there to check if the body's changed before redrawing. Sprite drawing code should check whether a species is genderless and determine the correct sprite based on that. Can I get a list of non-working species.

> lizards have no mutantcolor

I'll take a look at this too. I'd wager it's an issue with mutantcolor being set wrong or unset.

**Comment by Koriath** [comment]
No, lizards do get the colour, but only on the tail, and only when facing north. The rest of the sprite stays normal. Tha'ts what I found to be weird.
Curently, it's:
Shadowling (not shadow, the gamemode one)
Lizard colour bug (tail works North)
pod people
abductor
Plasmamen
These are bugged. Plasmamen I haven't finished the sprites, but Pod and Abductors for instance have the sprites there, but simply don't use them.

**Comment by Koriath** [comment]
Did a quick test. If I set a human to "neuter" gender and turn "gender_ambiguous" to 1, it draws the arms and legs for meeseeks. But it only draws arms for abductors. Draws nothing for Pod/Plant people either way. That's weird. It's like it stops drawing them for some reason.

**Comment by lkotipal** [comment]
I think I have a working fix. Pods, slimes and jellies still need greyscale sprites though
